src/ADAL.PCL/AuthenticationContext.cs

```diff
@@ -85,103 +92,118 @@ public AuthenticationContext(string authority, TokenCache tokenCache)
-            }
+            set { Authenticator.CorrelationId = value; }
         }
-             
+
         /// <summary>
-        /// Acquires device code from the authority.
+        ///     Acquires device code from the authority.
         /// </summary>
         /// <param name="resource">Identifier of the target resource that is the recipient of the requested token.</param>
         /// <param name="clientId">Identifier of the client requesting the token.</param>
         /// <returns>It contains Access Token, Refresh Token and the Access Token's expiration time.</returns>
         public async Task<DeviceCodeResult> AcquireDeviceCodeAsync(string resource, string clientId)
         {
-            return await this.AcquireDeviceCodeAsync(resource, clientId, null).ConfigureAwait(false);
+            return await AcquireDeviceCodeAsync(resource, clientId, null).ConfigureAwait(false);
         }
 
         /// <summary>
-        /// Acquires device code from the authority.
+        ///     Acquires device code from the authority.
         /// </summary>
         /// <param name="resource">Identifier of the target resource that is the recipient of the requested token.</param>
         /// <param name="clientId">Identifier of the client requesting the token.</param>
-        /// <param name="extraQueryParameters">This parameter will be appended as is to the query string in the HTTP authentication request to the authority. The parameter can be null.</param>
+        /// <param name="extraQueryParameters">
+        ///     This parameter will be appended as is to the query string in the HTTP authentication
+        ///     request to the authority. The parameter can be null.
+        /// </param>
         /// <returns>It contains Access Token, Refresh Token and the Access Token's expiration time.</returns>
-        public async Task<DeviceCodeResult> AcquireDeviceCodeAsync(string resource, string clientId, string extraQueryParameters)
+        public async Task<DeviceCodeResult> AcquireDeviceCodeAsync(string resource,
+            string clientId,
+            string extraQueryParameters)
         {
-            var handler = new AcquireDeviceCodeHandler(this.Authenticator, resource, clientId, extraQueryParameters);
-            return await handler.RunHandlerAsync().ConfigureAwait(false);
+            string requestId = Telemetry.GetInstance().CreateRequestId();
+            Telemetry.GetInstance().StartEvent(requestId, "api_event");
+
+            DeviceCodeResult result = null;
+            AcquireDeviceCodeHandler handler = new AcquireDeviceCodeHandler(Authenticator, resource, clientId,
+                extraQueryParameters, requestId);
+            try
+            {
+                result = await handler.RunHandlerAsync().ConfigureAwait(false);
+            }
+            finally
+            {
+                ApiEvent apiEvent = new ApiEvent(Authenticator, null, null, EventConstants.AcquireDeviceCodeAsync);
```

Comment:
Not necessarily.I am having the start event before the API call. And for the ApiEvent , the parameters are available after the acquireToken call is made. Hence it is after the call.